### PR TITLE
Pass RouteSettings to the internal Route in showCupertinoModalPopup

### PR DESCRIPTION
**Pull request by miquelbeltran**
## Description

In this PR I add the missing `RouteSettings` parameter to the `showCupertinoModalPopup` method.

This parameter already existed in the internal `_CupertinoModalPopupRoute` but it was not exposed.

It is useful to expose this parameter to provide the route name to the newly created `Route`.

As example, the Material `showDialog` function accepts `RouteSettings` as well.

## Related Issues

No related issues.

## Tests

Added a test that obtains the route name from the `RouteSettings` passed to the `showCupertinoModalPopup` call and compares it to the expected value, to verify that the value is correctly passed.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by miquelbeltran** [comment]
Is the Flutter team going to take a look at this? Because I don't want to
put the effort on keeping a PR updated if it is going to be ignored for
months.

**Comment by miquelbeltran** [comment]
Hi @goderbauer, is the a change I can get a review? the code is not a huge change but helps to 1:1 the Cupertino and Material APIs

File: packages/flutter/lib/src/cupertino/route.dart [hunk 1]
```diff
@@ -1035,6 +1038,7 @@ Future<T> showCupertinoModalPopup<T>({
   ImageFilter filter,
   bool useRootNavigator = true,
   bool semanticsDismissible,
+  RouteSettings settings,
```
Comment (goderbauer) [inline packages/flutter/lib/src/cupertino/route.dart:1041]: To keep the naming consistent with the other `showFoo` methods, can we also call this here `routeSettings`?

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**